datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -400,7 +403,7 @@ def build_where_clause(fltr):
             while f:
                 prop, value = f.popitem()
 
-                if isinstance(value, (tuple, list)):
+                if isinstance(value, (tuple, list)) and len(value) == 2 and isinstance(value[0], string_types):
```

Comment:
this is just horrible

Reply:
Yeah, and the following code is even worse. Maybe we could remove the extra check for size == 2? The only reason for the next lines to raise an IndexOutOfBounds was with lists of tuples. List of strings should always have two elements.